src/datasource.js

```diff
@@ -9,10 +9,19 @@ export class HawkularDatasource {
     this.type = instanceSettings.type;
     this.url = instanceSettings.url;
     this.name = instanceSettings.name;
-    this.tenant = instanceSettings.jsonData.tenant;
-    this.token = instanceSettings.jsonData.token;
     this.q = $q;
     this.backendSrv = backendSrv;
+    this.headers = {
+      'Content-Type': 'application/json',
+      'Hawkular-Tenant': instanceSettings.jsonData.tenant
+    };
+    if (typeof instanceSettings.basicAuth === 'string' && instanceSettings.basicAuth.length > 0) {
```

Comment:
Actually, not exactly. We now set the Authorization header by ourselves, whether it's direct mode or proxy mode (and btw that was already how we did for the bearer token).

However I will simplify that a little bit. We don't need to check both `instanceSettings.basicAuth` and `instanceSettings.jsonData.username`. I've checked other datasrouces such as influx, and checking just `instanceSettings.basicAuth` is enough.